pytorch_lightning/trainer/train_loader_patch.py

```diff
@@ -0,0 +1,82 @@
+'''
+    This patch solves two problems discussed in
+    https://github.com/PyTorchLightning/pytorch-lightning/pull/1959
+
+    The function  train_dataloader can either return a single instance of
+    torch.utils.data.DataLoader or a dictionary of dataloaders.
+
+    This patch fixes the length and iteration issus
+    and make the rest of the code oblivious of the underlying data structure.
+
+    I will keep the name of the class but a better name is probable advisable
+
+    @christofer-f

```

Comment:
why?

Reply:
Hi. This comment can be removed. It originates from the discussion when I suggested that the length of the data loader problem could be solved by a "magic" class. So in the first commit the class name was "magic_class" which is a bit silly 

:-)